src/kbmod/image_collection.py

```diff
@@ -0,0 +1,510 @@
+            Standardizer.
+
+        Returns
+        -------
+        std : `list``
+            A list of dictionaries containing the standardizer (``std``) and
+            the extension (``ext``) that maps to the given metadata row index.
+        """
+        if isinstance(idxs, int):
+            return [
+                self.get_standardizer(idxs, **kwargs),
+            ]
+        else:
+            return [self.get_standardizer(idx, **kwargs) for idx in idxs]
+
+    ########################
+    # FUNCTIONALITY (object operations, transformative functionality)
+    ########################
+    def write(self, *args, format=None, serialize_method=None, **kwargs):
+        """Write the ImageCollection to a file or file-like object.
+
+        A light wrapper around the underlying AstroPy's Table ``write``
+        functionality. See `astropy/io.ascii.write`
+        `documentation <https://docs.astropy.org/en/stable/io/ascii/write.html#parameters-for-write>`_
+        """
+        tmpdata = self.data.copy()
+
+        # a long history: https://github.com/astropy/astropy/issues/4669
+        # short of which is that WCS will not roundtrip itself the way we want
+        wcs_strs = []
+        for wcs in self.wcs:
+            header = wcs.to_header(relax=True)
+            h, w = wcs.pixel_shape
+            header["NAXIS1"] = (h, "height of the original image axis")
+            header["NAXIS2"] = (w, "width of the original image axis")
+            wcs_strs.append(header.tostring())
+        tmpdata["wcs"] = wcs_strs
+
+        bbox = [json.dumps(b) for b in self.bbox]
+        tmpdata["bbox"] = bbox
+
+        configs = [json.dumps(entry["std"].config.toDict()) for entry in self.standardizers]
+        # If we name this config then the unpacking operator in get_std will
+        # catch it. Otherwise, we need to explicitly handle it in read.
+        tmpdata["config"] = configs
+
+        # some formats do not officially support comments, like CSV, others
+        # have no problems with comments, some provide a workaround like
+        # dumping only meta["comment"] section if comment="#" kwarg is given.
+        # Because of these inconsistencies we'll just package everything into
+        # "comments" tag and then unpack at read time.
+        stringified = json.dumps(tmpdata.meta)
+        current_comments = tmpdata.meta.get("comments", None)
+        if current_comments is not None:
+            tmpdata.meta = {}
+        tmpdata.meta["comments"] = [
+            stringified,
+        ]
+
+        tmpdata.write(*args, format=format, serialize_method=serialize_method, **kwargs)
+
+    def get_zero_shifted_times(self):
+        """Returns a list of timestamps such that the first image
+        is at time 0.
+
+        Returns
+        -------
+        List of floats
+            A list of zero-shifted times (JD or MJD).
+        """
+        # what's the deal here - are we required to be sorted?
```

Comment:
It was assumed in the original KBMOD code. We should move off that, but we'd need to change things throughout.

Reply:
It wouldn't be that big of a deal to sort the table in `ImageCollection.__init__` or in `ImageCollection.fromStandardizer`. 

Former is easier as we just blindly always sort, but we pay that cost all the time, `fromStandardizer` require us to just be a bit careful about inserting things into the table. [Reading up on it](https://numpy.org/doc/stable/reference/generated/numpy.sort.html), it seems that while there are no guarantees, usually one should expect a smaller cost of sorting for an already sorted list.

Do you have any preferences?

Reply:
Let's always sort then.